AllReadyApp/Web-App/AllReady/Areas/Admin/Controllers/CampaignController.cs

```diff
@@ -117,11 +119,16 @@ public IActionResult Create()
                         ModelState.AddModelError("ImageUrl", "You must upload a valid image file for the logo (.jpg, .png, .gif)");
                         return View(campaign);
                     }
+
+                    if (existingImageUrl != null)
+                    {
+                        await _imageService.DeleteImageAsync(existingImageUrl);
+                    }
                 }

```

Comment:
We should probably enclose this delete code block within the if (fileUpload.IsAcceptableImageContentType()) and only on successfully getting a new imageUrl back from the imageService? If for any reason that fails to upload or is not the right content type we don't want to delete the existing image. Currently I believe this will fall through and delete the original image regardless of whether a new one is correctly uploaded.